src/garage/tf/algos/td3.py

```diff
@@ -376,14 +377,15 @@ def _optimize_policy(self, itr):
             float: Q value predicted by the q network.
 
         """
-        transitions = self._replay_buffer.sample_transitions(
+        timesteps = self._replay_buffer.sample_timesteps(
             self._buffer_batch_size)
 
-        observations = transitions['observations']
-        rewards = transitions['rewards']
-        actions = transitions['actions']
-        next_observations = transitions['next_observations']
-        terminals = transitions['terminals']
+        observations = timesteps.observations
+        rewards = timesteps.rewards.reshape(-1, 1)
+        actions = timesteps.actions
+        next_observations = timesteps.next_observations
+        terminals = np.array([[s == StepType.TERMINAL]
+                              for s in timesteps.step_types])
```

Comment:
is there a reason you unpack all of these int aliases? IMO the most idiomatic usage would be to pass the entire TimeStepBatch data structure to TF (i.e. to self._train_qf). At least that's the idea.